netsnmp/__init__.py

```diff
@@ -60,14 +60,13 @@ def oid_outside_tree(oid1, oid2):
     Return True if oid2 outside of oid1 tree
     Used by SNMPSession.walk
     """
-    oid1_split = [i for i in oid1.split('.') if i]
-    oid1_idx = len(oid1_split)-1
-
-    oid2_split = [i for i in oid2.split('.') if i]
-
-    if oid2_split[oid1_idx] > oid1_split[oid1_idx]:
+    try:
+        if oid2.startswith(oid1):
+            return False
+        return True
+    except:
         return True
-    return False
+
```

Comment:
You are correct. I guess this is a discrepancy between the netsnmp app functionality and this python module.

Our walk() function will work if you pass a full OID to walk, as all it does it perform getnext()

Given that, we could change walk(), but right now I'd prefer to leave it as is, since it should work either way currently.